src/mappings/purchaseOffer.ts

```diff
@@ -59,9 +59,17 @@ export async function handlePurchaseOfferAccepted(
   assert(event.args, 'No event args');
 
   const eventOfferId = event.args.offerId.toString();
+  const eventAgreementId = event.args.agreementId.toString();
+
   const offer = await Offer.get(eventOfferId);
   assert(offer, `offer not found. offerID="${eventOfferId}"`);
 
+  const serviceAgreement = await ServiceAgreement.get(eventAgreementId);
```

Comment:
Better not check this.  Typically `OfferAcceptedEvent` should behind the `ServiceAgreementCreated` event, there still have possibility that when the assert execute but the new agreement is not saved yet even though they are in the same transaction.